mantidproject/mantid
# Filtering not applied in load processed nexus

**Description of work.**

Applies filtering after loading a processed nexus file, includes a unit test to check this for a single workspace. The instructions below will also check multi period files.
All of the changes in this are framework only and should work equally well in both interfaces.

**To test:**
In mantidplot:
1. Load POLREF14699 (in unit test data)
1. Save the groupWorkspace out to "testMulti.nxs"
1. Save the first member of the group to "testSingle.nxs"
1. Load both of those into different workspaces
1. Open the sample logs view for all of the three copies of the same workspace (original, saved singly and as a groupws).
1. Look at the same log in each (I used raw_uah_log as it changes during the expt).
1. Flip the filtering between the various options and look at the stats change, check they are the same across all of the workspaces. (in workbench the filtering is always Status + Period).
1. Note one of the stats that changes e,g, Max = 193.333 when filtered

Open workbench
1. Load the two test files
1. Check that max value on the two copies of the same workspace.  It should match the value from Mantidplot.

Fixes #28572


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
